cpp/src/copying/scatter.cu

```diff
@@ -172,27 +174,29 @@ struct column_scalar_scatterer_impl<dictionary32, MapIterator> {
                                      MapIterator scatter_iter,
                                      size_type scatter_rows,
                                      column_view const& target,
-                                     rmm::mr::device_memory_resource* mr,
-                                     cudaStream_t stream) const
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr) const
   {
     auto dict_target = dictionary::detail::add_keys(
       dictionary_column_view(target),
-      make_column_from_scalar(source.get(), 1, rmm::mr::get_current_device_resource(), stream)
+      make_column_from_scalar(source.get(), 1, stream, rmm::mr::get_current_device_resource())
         ->view(),
       mr,
-      stream);
+      stream.value());
     auto dict_view    = dictionary_column_view(dict_target->view());
     auto scalar_index = dictionary::detail::get_index(
-      dict_view, source.get(), rmm::mr::get_current_device_resource(), stream);
+      dict_view, source.get(), stream, rmm::mr::get_current_device_resource());
```

Comment:
mr is a default param now.

Reply:
I don't understand this comment. `dictionary::detail::get_index()` has no default parameters.

Reply:
https://github.com/rapidsai/cudf/pull/6648/files?file-filters%5B%5D=.hpp#diff-da300662b4bc48b59f81fbba20725eacf62dbabb0a0c671e2fa8b05aeefca775R32-R36
 Similar suggestion as https://github.com/rapidsai/cudf/pull/6648#discussion_r523259260 of not needing mr.

Reply:
OK, you convinced me. I changed your two suggestions and all the others I could find to not call `get_current_device_resource` when that is the default.